js/src/MapObject.js

```diff
@@ -125,13 +127,39 @@ MapObject.prototype.add_annotation = function (type, args, kwargs) {
     this._color_counter++;
   }
 
+  var style = {
+    fillColor: kwargs.rgb || this.next_color(),
+    fillOpacity: 0.8,
+    strokeWidth: 2
+  };
+
+  /*
+   * This is an ugly hack to get around the fact that geojs doesn't return
+   * the created annotation object.  Currently, the only way to get this
+   * object is to listen to an event that also fires when generating annotations
+   * with the mouse.  Further, the events are not fired consistently:
+   *
+   *  * When creating an annotation with the mouse, it is only triggered with state
+   *    with mode `create` before it has any coordinate information.  No event
+   *    add event is triggered after the annotation is finished.
+   *  * When creating via the API, it is only triggered with state `done`.
+   *
+   * The code here expects this event will be triggered synchronously.  This is
+   * currently true, but may change in the future.
+   */
```

Comment:
Yes, but given current priorities, I wouldn't expect it soon.  A quick solution that would break backward compatibility would be to return the annotation rather than the layer from `createAnnotation`.  A more long term goal would be to ensure events are more consistently fired and documented.